recipe/meta.yaml

```diff
@@ -29,18 +38,22 @@ requirements:
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
     - cython                                 # [build_platform != target_platform]
     - numpy                                  # [build_platform != target_platform]
-    - {{ compiler('c') }}  # [unix]
-    - {{ compiler('cxx') }}  # [unix]
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
   host:
     - python
     - pip
-    - cython             # [unix]
-    - arrow-cpp  0.17.1  # [unix]
-    - pyarrow  0.17.1    # [unix]
-    - numpy              # [unix]
+    - cython
+    - arrow-cpp
+    - pyarrow
+    - numpy
   run:
     - python
-    - pyarrow 0.17.1
+    {% if pyarrow == '0.17.1' %}
+    - pyarrow {{ pyarrow }}
+    {% else %}
+    - {{ pin_compatible('pyarrow', max_pin='x') }}
```

Comment:
this seems to put a pin for 3.0 in both cases??

Reply:
nm seems like a weird conda index bug locally

Reply:
Rebuilt this locally and the pins looks fine, merging this.